src/types/list.rs

```diff
@@ -592,14 +594,14 @@ impl<'a> IntoIterator for &'a PyList {
 }
 
 /// Used by `PyList::iter()`.
-pub(crate) struct PyListIterator2<'py> {
-    list: Py2<'py, PyList>,
+pub struct PyListIterator2<'py> {
```

Comment:
This one should also loose it `2` in exchange for a `Bound` somewhere.

Reply:
Too me, `Bound/BorrowFooIterator` seem the most reasonable of the choices.

Reply:
`BoundListIterator` it is 👍